lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1223,21 +1223,197 @@ def test__lazy(self):
         self._check_copy(cube, cube.copy())
 
 
+class Test_fill_value(tests.IrisTest):
+    def setUp(self):
+        self.dtypes = (np.dtype('int'), np.dtype('uint'),
+                       np.dtype('bool'), np.dtype('float'))
+
+    def test_cube___init___default_none(self):
+        cube = Cube(np.array(0))
+        self.assertIsNone(cube.fill_value)
+
+    def test_cube___init___setter(self):
+        fill_value = 123
+        cube = Cube(np.array(0), fill_value=fill_value)
+        self.assertEqual(cube.fill_value, fill_value)
+
+    def test_cube_setter(self):
+        fill_value = 123
+        cube = Cube(np.array(0))
+        self.assertIsNone(cube.fill_value)
+        cube.fill_value = fill_value
+        self.assertEqual(cube.fill_value, fill_value)
+
+    def test_cube_data_setter_with_none(self):
+        fill_value = 123
+        cube = Cube(np.array(0), fill_value=fill_value)
+        self.assertEqual(cube.fill_value, fill_value)
+        cube.data = np.array(1)
+        self.assertEqual(cube.fill_value, None)
+
+    def test_fill_value_bool(self):
+        fill_value = np.array([1], dtype=np.dtype('bool'))[0]
```

Comment:
It would have to be `fill_value = np.array(True)[()]` which I thought was a little bit terse and magical ... and this ain't Hogwarts, so I erred on the side of verbosity :wink: